### PR TITLE
Improve match for `DbCamera_CalcUpFromPitchYawRoll`, add docs

### PR DESCRIPTION
**Pull request by engineer124**
Very similar to `Camera_CalcUpFromPitchYawRoll`, the only differences being:
- No pad
- `baseUp` wants to be at the top of the function
- Extra `(void)0, ` on part of the matrix

### CODE REVIEW

**Review by Dragorn421 — APPROVED** [review]
the `(void)0,` is ugly but oh well

**Comment by engineer124** [comment]
> the `(void)0,` is ugly but oh well

Messed around with it a bit more, found a solution without the `(void)0,`

**Review by Dragorn421 — APPROVED** [review]
heeey, my hero

File: src/code/db_camera.c [hunk 1]
```diff
@@ -109,61 +109,54 @@ Vec3f* DbCamera_AddVecSph(Vec3f* out, Vec3f* in, VecSph* sph) {
...
+    rollMtxRow2.z = ((1.0f - cosR) * (u.y * u.z)) - (u.x * sinR);
+
+    rollMtxRow3.x = ((1.0f - cosR) * (u.z * u.x)) - (u.y * sinR);
+    rollMtxRow3.y = ((1.0f - cosR) * (u.y * u.z)) + (u.x * sinR);
+    rollMtxRow3.z = ((1.0f - SQ(u.z)) * cosR) + SQ(u.z);
+
+    // rollMtx * baseUp
+    up.x = DOTXYZ(baseUp, rollMtxRow1);
+    up.y = DOTXYZ(baseUp, rollMtxRow2);
+    up.z = DOTXYZ(baseUp, rollMtxRow3);
```
Comment (EllipticEllipsis) [inline src/code/db_camera.c:155]: I still don't think this is semantically correct for matrix multiplication.

**Review by fig02 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

File: src/code/db_camera.c (hunk 1 above)
Comment (Dragorn421) [reply src/code/db_camera.c:155]: @EllipticEllipsis can we merge this or is this too big of a deal?
Me I'm not looking at the dotxyz macro further than "sum the xx yy zz products" and just as a space-saving / readability-helper macro

**Pull request merged**